src/utils/middlewares/azure_user_attributes_manage.ts

```diff
@@ -86,7 +101,43 @@ export const AzureUserAttributesManageMiddleware = (): IRequestMiddleware<
 
   if (subscriptionId.startsWith("MANAGE-")) {
     // MANAGE Flow
+    const errorOrMaybeAuthorizedCIDRs = await subscriptionCIDRsModel.findLastVersionByModelId(
+      [subscriptionId]
+    )();
+
+    if (E.isLeft(errorOrMaybeAuthorizedCIDRs)) {
+      winston.error(
+        `No CIDRs found for subscription|${subscriptionId}|${JSON.stringify(
+          errorOrMaybeAuthorizedCIDRs.left
+        )}`
+      );
+      return E.left<
+        IResponse<"IResponseErrorQuery">,
+        IAzureUserAttributesManage
+      >(
+        ResponseErrorQuery(
+          `Error while retrieving CIDRs tied to the provided subscription id`,
+          errorOrMaybeAuthorizedCIDRs.left
+        )
+      );
+    }
+
+    const maybeAuthorizedCIDRs = errorOrMaybeAuthorizedCIDRs.right;
+
+    if (isNone(maybeAuthorizedCIDRs)) {
+      winston.error(
+        `IAzureUserAttributesManage|CIDRs not found|${subscriptionId}`
+      );
+      return E.left<
```

Comment:
We are stating that a manage key MUST have IP protection. Is this correct?

Reply:
Yes, it's our security goal for this development.
In addition, it should be noted that when we will create a Subscription _(MANAGE)_, we will also create a starting value for subscription related cidrs _(empty array as for services auth_cidrs)._

Reply:
Uhm, I am not sure we need to enforce such a rule. We do not, so far, with actual API keys.

I think we are solving the problem:

> Users that currently **decided to use** IP protection, want to do the same with manage keys

which is slightly different than:

> Users that will use manage keys will be **required to use** IP protection

----

Moving a bit forward: should we enforce users to provide IP protection? Is it good for the product? We can discuss it (cc @iwoak)